frontend/public/components/utils/csv-watch-hook.ts

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { useK8sWatchResource } from './k8s-watch-hook';
+import {
+  ClusterServiceVersionModel,
+  ClusterServiceVersionKind,
+} from '@console/operator-lifecycle-manager';
+import { referenceForModel } from '@console/internal/module/k8s';
+
+export const useCsvWatchResource = (ns: string) => {
+  const csvResource = React.useMemo(
+    () => ({
+      isList: true,
+      kind: referenceForModel(ClusterServiceVersionModel),
+      namespace: ns,
+      optional: true,
+    }),
+    [ns],
+  );
+
+  const [csvData] = useK8sWatchResource<ClusterServiceVersionKind[]>(csvResource);
+  return csvData;
```

Comment:
should add a check for the data loaded and error here

Reply:
adding a check for error wouldn't make much difference here, if the csvData is undefined it wouldn't break the UI as the util to check if a workload is operator backed takes care of it. Anyway I have updated the csv-watch-hook to return `csvLoaded & csvError` as well